cep-20.2.md

```diff
@@ -0,0 +1,829 @@
+`recipe` is also merged into each output. The top-level name is ignored (but could be used, e.g. for
+the feedstock-name in the conda-forge case).
+
+```yaml
+# note: instead of `package`, it says `recipe` here
+recipe:
+  name: string # mostly ignored, could be used for feedstock name
+  version: string # merged into each output if not overwritten by output
+
+outputs:
+  - package:
+      name: string
+      version: string (defaults to top-level version)
+    build:
+      script: ...
+    requirements:
+      # same definition as top level
+
+    # same definitions as on top level, by default merged from outer recipe
+    about:
+    source:
+    test:
+```
+
+Before the build, the outputs are topologically sorted by their dependencies. Each output acts as an independent recipe.
+
+> **Note**
+> A previous version contained a idea for a "cache-only" output. We've moved that to a future CEP.
+
+### Aside: variant computation for multiple outputs
+
+Multiple outputs are treated like individual recipes after the merging of the nodes is completed.
+Therefore, each variant is computed for each output individually.
+
+Another tricky bit is that packages can be "strongly" connected with a `pin_subpackage(name, exact=True)` constraint.
+In this case, the pinned package should also be part of the "variant configuration" for the output and appropriately zipped.
+
+For example, we can have three outputs: `libmamba`, `libmambapy`, and `mamba`.
+
+- `libmamba` -> creates a single variant as it is a low-level C++ library
+- `libmambapy` -> creates multiple packages, one per python version
+- `mamba` -> creates multiple packages (one per python + libmambapy version)
+
+<img width="1527" alt="Screenshot 2023-11-01 at 15 48 59" src="https://github.com/wolfv/ceps/assets/885054/3ddb7ddb-f563-4842-91d6-d9f8c91b83e0">
+
+This has historically been an issue in conda-build, as it didn't take into account `pin_subpackage` edges as "variants" and
+sometimes created the same hashes for two different outputs. E.g. in the following situation, conda-build would only create a single
+`foofoo` package (instead of two variants that exactly pin two different libfoo packages):
+
+<img width="2065" alt="Screenshot 2023-11-01 at 15 47 49" src="https://github.com/wolfv/ceps/assets/885054/0c3ec263-0d00-45a7-b6fc-84e9b8daffa6">
+
+## About section
+
+```yaml
+about:
+  # a summary of what this package does
+  summary: string
+  # a longer description of what this package does (should we allow referencing files here?)
+  description: string
+
+  # the license of the package in SPDX format
+  license: string (SPDX enforced)
+  # the license files
+  license_file: path | [path] (relative paths are found in source directory _or_ recipe directory)
+  # URL that points to the license – just stored as metadata
+  license_url: url
+
+  # URL to the homepage (used to be `home`)
+  homepage: url
+  # URL to the repository (used to be `dev_url`)
+  repository: url
```

Comment:
We based this off of cargo. It can be left out if there is no repository.

Reply:
people can also still use the "homepage" url...

Reply:
In any case, I think we should still retain the 3 home/dev/doc (no matter the naming) since it is common for them to be distinct and/or have overlaps between any combination of them.

Reply:
I'd stick to development too.